contracts/modules/escrow/Escrow.sol

```diff
@@ -73,17 +73,18 @@ contract Escrow is IEscrow {
         uint collateralValue = 0;
         for(uint i = 0; i < _tokensUsedAsCollateral.length; i++) {
             address token = _tokensUsedAsCollateral[i];
-            uint deposit = deposited[token];
+            uint deposit = deposited[token].amount;
             if(deposit != 0) {
-                uint price = IOracle(oracle).getLatestAnswer(token);
-                // need to scale value by token decimal
-                (bool success, bytes memory result) = token.call(abi.encodeWithSignature("decimals()"));
-                if(!success) {
-                    collateralValue += (price * deposit) / 1e18;
-                } else {
-                    uint8 decimals = abi.decode(result, (uint8));
-                    collateralValue += (price * deposit) / (1 * 10 ** decimals);
+                if(deposited[token].isERC4626) {
+                    // this conversion could shift, hence it is best to get it each time
+                    (bool success, bytes memory assetAmount) = token.call(abi.encodeWithSignature("previewRedeem(uint256)", deposit));
```

Comment:
@kibagateaux this is the most accurate way to calculate the tokens you would get but that doesn't mean that there won't be variation when you actually redeem it e.g. on liquidation.